packages/cashier/src/components/cashier-search-box/cashier-search-box.jsx

```diff
@@ -0,0 +1,74 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Field as FormField, Formik, Form } from 'formik';
+import { Icon, Input } from '@deriv/components';
+import './cashier-search-box.scss';
+
+const CashierSearchBox = ({ className, onClear, onSearch, placeholder, setIsSearchLoading }) => {
+    const onSearchClear = setFieldValue => {
+        setFieldValue('search', '');
+        if (typeof onClear === 'function') {
+            onClear();
+        }
+    };
+
+    const onSearchKeyUpDown = (submitForm, search) => {
+        if (!search.trim()) return;
+        setIsSearchLoading(true);
+        clearTimeout(typing_timer);
+        const typing_timer = setTimeout(() => {
+            submitForm();
+        }, 300);
+    };
+
+    const onSearchSubmit = ({ search }) => {
+        if (typeof onSearch === 'function') {
+            onSearch(search);
+        }
+    };
+
+    return (
+        <div className={classNames('cashier-search-box', className)}>
```

Comment:
Ideally yes, but sometimes designers can change something in design and just in case, I left a way to change the style of the component, also we have many reusable components in components package which have the same approach.